ironfish/src/utils/percentile.ts

```diff
@@ -0,0 +1,49 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
```

Comment:
This looks like a lot of sorting logic as well. If this is not going to be used in any performance critical paths then maybe just something like this would suffice and be easier to maintain
```typescript
function percentile(data: number[], percentile: number):  number | null {
	const sorted = data.slice()
	sorted.sort()
	const k = Math.ceil((data.length * percentile) / 100) - 1
	return data[k]
}
```

I think for getting multiple percentiles in a row it could be faster to sort the whole thing upfront too before getting the kth element (if performance is an issue). Don't think that's necessary though

Reply:
definitely agree here. For simplicity and optimization, would prefer to see an approach in which you just sort all the numbers, calculate the 25/50/75 percentile indices, then just pick from your sorted array. 

Wouldn't really even need its own util, and could be a function in the main RPC code.

Reply:
yea good point, shouldn't be sorting multiple times. removed util and simplified this function.